include/language.php

```diff
@@ -148,78 +180,234 @@ class PLL_Language {
+			'count'               => array( 'language', 'count' ),
+			'tl_term_id'          => array( 'term_language', 'term_id' ),
+			'tl_term_taxonomy_id' => array( 'term_language', 'term_taxonomy_id' ),
+			'tl_count'            => array( 'term_language', 'count' ),
+		);
+
+		// Deal with duplicated data: priority to the ones in `$this->term_props`.
+		foreach ( $term_props as $prop => $value ) {
+			if ( isset( $language_data['term_props'][ $value[0] ][ $value[1] ] ) ) {
+				$this->$prop = $this->set_tax_prop( $value[0], $value[1], $language_data['term_props'][ $value[0] ][ $value[1] ] );
+			} elseif ( isset( $language_data[ $prop ] ) ) {
+				$this->$prop = $this->set_tax_prop( $value[0], $value[1], $language_data[ $prop ] );
+			}
+
+			unset( $language_data['term_props'][ $value[0] ][ $value[1] ], $language_data[ $prop ] );
+		}
+
+		// Other values in `term_props` that are not `language` nor `term_language`.
+		if ( ! empty( $language_data['term_props'] ) ) {
+			foreach ( $language_data['term_props'] as $taxonomy_name => $prop_values ) {
+				foreach ( $prop_values as $prop_name => $prop_value ) {
+					$this->set_tax_prop( $taxonomy_name, $prop_name, $prop_value );
+				}
+			}
+		}
+
+		unset( $language_data['term_props'] );
+
+		// Add all the other values.
+		foreach ( $language_data as $prop => $value ) {
+			$this->$prop = $value;
+		}
+	}
+
+	/**
+	 * Returns a language term property value (like term ID, term taxonomy ID, or count).
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $taxonomy_name Name of the taxonomy.
+	 * @param string $prop_name     Name of the property: 'term_taxonomy_id', 'term_id', 'count'.
+	 * @return int
+	 *
+	 * @phpstan-param non-empty-string $taxonomy_name
+	 * @phpstan-param non-empty-string $prop_name
+	 * @phpstan-return int<0, max>
+	 */
+	public function get_tax_prop( $taxonomy_name, $prop_name ) {
+		return isset( $this->term_props[ $taxonomy_name ][ $prop_name ] ) ? $this->term_props[ $taxonomy_name ][ $prop_name ] : 0;
+	}
+
+	/**
+	 * Stores a language term property value (like term ID, term taxonomy ID, or count).
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $taxonomy_name Name of the taxonomy.
+	 * @param string $prop_name     Name of the property: 'term_taxonomy_id', 'term_id', 'count'.
+	 * @param int    $prop_value    Property value.
+	 * @return int
+	 *
+	 * @phpstan-param non-empty-string $taxonomy_name
+	 * @phpstan-param non-empty-string $prop_name
+	 * @phpstan-return int<0, max>
 	 */
-	public function __construct( $language, $term_language = null ) {
-		if ( empty( $term_language ) ) {
-			// Build the object from all properties stored as an array.
-			foreach ( $language as $prop => $value ) {
-				$this->$prop = $value;
+	public function set_tax_prop( $taxonomy_name, $prop_name, $prop_value ) {
+		if ( ! is_numeric( $prop_value ) ) {
+			return 0;
+		}
+
+		$prop_value = $prop_value >= 1 ? abs( (int) $prop_value ) : 0;
```

Comment:
Why use `abs()`? I don't understand 😅

Reply:
So `$prop_value` is considered as a 0+ integer by PHPStan 😉.

Reply:
Ok! Nice trick